sdk/src/metrics/meter.cc

```diff
@@ -293,6 +295,7 @@ std::unique_ptr<AsyncWritableMetricStorage> Meter::RegisterAsyncMetricStorage(
 std::vector<MetricData> Meter::Collect(CollectorHandle *collector,
                                        opentelemetry::common::SystemTimestamp collect_ts) noexcept
 {
+  std::lock_guard<opentelemetry::common::SpinLockMutex> guard(storage_lock_);
```

Comment:
Move this to below context lock? So the error handling and call to `Observe` is not necessary to be guarded by this lock?

Reply:
Yes, good point.  Have moved it below the context lock.